Rakefile

```diff
@@ -18,6 +18,7 @@ LOG = Logger.new(STDERR)
 
 # Indicates whether this is being run from within a 'search:' task in rake.
 RAKE_SEARCH = (Rake.application.top_level_tasks.select {|task| task.include? 'search:'}).any?
+RAKE_SEARCH_INITIALIZE = (Rake.application.top_level_tasks.select {|task| task.include? 'search:initialize'}).any?
```

Comment:
I was wondering if there was any way we could merge these so that it didn't have to be two different things.